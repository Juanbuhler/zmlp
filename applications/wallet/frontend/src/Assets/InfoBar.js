import NavbarWrapper from '../Navbar/Wrapper'

import { constants, colors } from '../Styles'

const BOX_SHADOW = '0px 3px 3px 0 rgb(0, 0, 0, 0.3)'

const AssetInfoBar = () => {
  // Temporary values
  const displayCount = 100
  const totalCount = 2400

  return (
    <NavbarWrapper
      style={{
        top: constants.navbar.height + 1,
        justifyContent: 'flex-end',
        boxShadow: BOX_SHADOW,
      }}>
      <div css={{ color: colors.structure.steel }}>
        Sort: Import Date &nbsp;| &nbsp;
      </div>
      <div css={{ color: colors.structure.zinc }}>
        {displayCount} of {totalCount} Results
      </div>
    </NavbarWrapper>
  )
}

export default AssetInfoBar
