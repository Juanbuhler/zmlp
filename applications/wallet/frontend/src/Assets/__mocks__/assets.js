const assets = {
  count: 574,
  next: 'https://wallet.zmlp.zorroa.com/api/v1/projects/03aa4218-1243-4a62-8113-42064dcf8f0e/searches/query/?from=100&size=100',
  previous: null,
  results: [
    {
      id: 'pNwnXjVntgbDQgPZhkXqVT-2URMqvJNL',
      metadata: {
        source: {
          path: 'gs://zmlp-private-test-data/zorroa-deploy-testdata/zorroa-dev-testdata/zorroa-test-data/office/test_document.docx',
          extension: 'docx',
          filename: 'test_document.docx',
          mimetype:
            'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
        },
        analysis: {
          'flower-types': {
            count: 1,
            type: 'labels',
            predictions: [{ score: 0.153, label: 'rose' }],
          },
        },
      },
      thumbnailUrl:
        'https://wallet.zmlp.zorroa.com/api/v1/projects/03aa4218-1243-4a62-8113-42064dcf8f0e/assets/pNwnXjVntgbDQgPZhkXqVT-2URMqvJNL/files/category/web-proxy/name/web-proxy.jpg/',
    },
    {
      id: '3DDnuCNruXiXtRjKxwGJt2UPGNTAZut4',
      metadata: {
        source: {
          path: 'gs://zmlp-private-test-data/zorroa-deploy-testdata/zorroa-dev-testdata/zorroa-test-data/office/irr.pdf',
          extension: 'pdf',
          filename: 'irr.pdf',
          mimetype: 'application/pdf',
        },
        analysis: {
          'flower-types': {
            count: 1,
            type: 'labels',
            predictions: [{ score: 0.153, label: 'rose' }],
          },
        },
      },
      thumbnailUrl:
        'https://wallet.zmlp.zorroa.com/api/v1/projects/03aa4218-1243-4a62-8113-42064dcf8f0e/assets/3DDnuCNruXiXtRjKxwGJt2UPGNTAZut4/files/category/web-proxy/name/web-proxy.jpg/',
    },
    {
      id: 'iKiUfpLIq3lNGu7YZoVDRu1eEUAG3Gbo',
      metadata: {
        source: {
          path: 'gs://zmlp-private-test-data/zorroa-deploy-testdata/zorroa-dev-testdata/zorroa-test-data/office/irr.pdf',
          extension: 'pdf',
          filename: 'irr.pdf',
          mimetype: 'application/pdf',
        },
        analysis: {
          'flower-types': {
            count: 1,
            type: 'labels',
            predictions: [{ score: 0.153, label: 'rose' }],
          },
        },
      },
      thumbnailUrl:
        'https://wallet.zmlp.zorroa.com/api/v1/projects/03aa4218-1243-4a62-8113-42064dcf8f0e/assets/iKiUfpLIq3lNGu7YZoVDRu1eEUAG3Gbo/files/category/web-proxy/name/web-proxy.jpg/',
    },
    {
      id: 'LDOJ879pyZn4Y-gRhQmd6q3h_kG8mM2r',
      metadata: {
        source: {
          path: 'gs://zmlp-private-test-data/zorroa-deploy-testdata/zorroa-dev-testdata/zorroa-test-data/office/irr.pdf',
          extension: 'pdf',
          filename: 'irr.pdf',
          mimetype: 'application/pdf',
        },
        analysis: {
          'flower-types': {
            count: 1,
            type: 'labels',
            predictions: [{ score: 0.153, label: 'rose' }],
          },
        },
      },
      thumbnailUrl:
        'https://wallet.zmlp.zorroa.com/api/v1/projects/03aa4218-1243-4a62-8113-42064dcf8f0e/assets/LDOJ879pyZn4Y-gRhQmd6q3h_kG8mM2r/files/category/web-proxy/name/web-proxy.jpg/',
    },
    {
      assetStyle: 'video',
      id: 'srL8ob5cTpCJjYoKkqqfa2ciyG425dGi',
      metadata: {
        source: {
          checksum: 2431356942,
          extension: 'mp4',
          filename: 'FatManOnABike1914.mp4',
          filesize: 13168719,
          mimetype: 'video/mp4',
          path: 'gs://zmlp-private-test-data/zorroa-deploy-testdata/zorroa-cypress-testdata/mp4/FatManOnABike1914.mp4',
        },
        analysis: {
          'flower-types': {
            count: 1,
            type: 'labels',
            predictions: [{ score: 0.153, label: 'rose' }],
          },
        },
      },
      thumbnailUrl:
        'https://wallet.zmlp.zorroa.com/api/v1/projects/03aa4218-1243-4a62-8113-42064dcf8f0e/assets/srL8ob5cTpCJjYoKkqqfa2ciyG425dGi/files/category/web-proxy/name/web-proxy.jpg/',
      videoLength: 144.45,
      videoProxyUrl:
        'https://wallet.zmlp.zorroa.com/api/v1/projects/03aa4218-1243-4a62-8113-42064dcf8f0e/assets/srL8ob5cTpCJjYoKkqqfa2ciyG425dGi/files/category/proxy/name/video_450x360.mp4/',
    },
  ],
}

export default assets
