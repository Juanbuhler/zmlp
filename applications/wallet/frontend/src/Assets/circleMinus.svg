<svg width="20" height="20" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><defs><path d="M14 10a1 1 0 01-1 1H7a1 1 0 010-2h6a1 1 0 011 1m-4 8c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8m0-18C4.477 0 0 4.477 0 10s4.477 10 10 10 10-4.477 10-10S15.523 0 10 0" id="a"/></defs><g fill="none" fill-rule="evenodd"><path d="M0 0h20v20H0z"/><use fill="#000" xlink:href="#a"/></g></svg>