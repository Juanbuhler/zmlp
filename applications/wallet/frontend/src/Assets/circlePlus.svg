<svg width="20" height="20" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><defs><path d="M11 9h3v2h-3v3H9v-3H6V9h3V6h2v3zm-1 9c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8zm0-18C4.477 0 0 4.477 0 10s4.477 10 10 10 10-4.477 10-10S15.523 0 10 0z" id="a"/></defs><use fill="#000" xlink:href="#a" fill-rule="evenodd"/></svg>