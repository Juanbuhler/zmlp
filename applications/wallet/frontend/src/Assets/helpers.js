export const formatSeconds = ({ seconds }) => {
  const ISOString = new Date(seconds * 1000).toISOString()

  // has double digit hours
  if (seconds > 36000) return ISOString.substr(11, 8)

  // has single digit hours
  if (seconds > 3600) return ISOString.substr(12, 7)

  // has double digit minutes
  if (seconds > 600) return ISOString.substr(14, 5)

  // has single digit minutes or less than 1 minute
  return ISOString.substr(15, 4)
}
