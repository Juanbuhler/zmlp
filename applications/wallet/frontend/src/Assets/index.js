import { useReducer } from 'react'
import { useRouter } from 'next/router'
import useSWR, { useSWRPages } from 'swr'

import { spacing } from '../Styles'

import Loading from '../Loading'

import AssetsResize from './Resize'
import AssetsThumbnail from './Thumbnail'

import { reducer, INITIAL_STATE } from './reducer'

const SIZE = 50

const Assets = () => {
  const {
    query: { projectId },
  } = useRouter()

  const [state, dispatch] = useReducer(reducer, INITIAL_STATE)

  const { thumbnailCount, isMin, isMax } = state

  const containerWidth = 100 / thumbnailCount

  const { pages, isLoadingMore, isReachingEnd, loadMore } = useSWRPages(
    // page key
    'visualizer',

    // page component
    ({ offset, withSWR }) => {
      const from = offset * SIZE
      const { data: { results } = {} } = withSWR(
        // eslint-disable-next-line react-hooks/rules-of-hooks
        useSWR(
          `/api/v1/projects/${projectId}/assets/?from=${from}&size=${SIZE}`,
          { suspense: false },
        ),
      )

      if (!results) {
        return (
          <div css={{ flex: 1, display: 'flex', height: '100%' }}>
            <Loading />
          </div>
        )
      }

      return results.map((asset) => (
        <AssetsThumbnail key={asset.id} asset={asset} />
      ))
    },

    // offset of next page
    ({ data: { count } }, index) => {
      const offset = (index + 1) * SIZE
      return offset < count ? offset : null
    },

    // deps of the page component
    [],
  )

  return (
    <div css={{ flex: 1, position: 'relative' }}>
      <div
        css={{
          height: '100%',
          display: 'flex',
          flexWrap: 'wrap',
          alignContent: 'flex-start',
          overflowY: 'auto',
          padding: spacing.small,
          '.container': {
            width: `${containerWidth}%`,
            paddingBottom: `${containerWidth}%`,
          },
        }}
      >
        {pages}
      </div>

      <AssetsResize dispatch={dispatch} isMin={isMin} isMax={isMax} />
    </div>
  )
}

export default Assets
