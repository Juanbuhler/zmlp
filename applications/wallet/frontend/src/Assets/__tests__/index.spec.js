import TestRenderer, { act } from 'react-test-renderer'

import Assets from '..'

import assets from '../__mocks__/assets'

import { encode } from '../../Filters/helpers'

const PROJECT_ID = '76917058-b147-4556-987a-0a0f11e46d9b'
const ASSET_ID = assets.results[0].id

jest.mock('../../Visualizer/Navigation', () => 'VisualizerNavigation')
jest.mock('../../Asset/Asset', () => 'AssetAsset')

describe('<Assets />', () => {
  it('should render properly while loading', () => {
    require('next/router').__setUseRouter({
      pathname: '/[projectId]/visualizer',
      query: { projectId: PROJECT_ID },
    })

    require('swr').__setMockUseSWRInfiniteResponse({})

    const component = TestRenderer.create(<Assets />)

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should render properly', () => {
    require('next/router').__setUseRouter({
      pathname: '/[projectId]/visualizer',
      query: { projectId: PROJECT_ID },
    })

    require('swr').__setMockUseSWRInfiniteResponse({ data: [assets] })

    const component = TestRenderer.create(<Assets />)

    expect(component.toJSON()).toMatchSnapshot()

    // Zoom in to 4 per row
    act(() => {
      component.root.findByProps({ 'aria-label': 'Zoom In' }).props.onClick()
    })

    // Zoom in to 1 per row
    act(() => {
      component.root.findByProps({ 'aria-label': 'Zoom In' }).props.onClick()
    })

    // Attempt to zoom at max size
    act(() => {
      component.root.findByProps({ 'aria-label': 'Zoom In' }).props.onClick()
    })

    expect(
      component.root.findByProps({ 'aria-label': 'Zoom In' }).props.isDisabled,
    ).toBe(true)
    expect(component.toJSON()).toMatchSnapshot()

    // Zoom out to 4 per row
    act(() => {
      component.root.findByProps({ 'aria-label': 'Zoom Out' }).props.onClick()
    })

    // Zoom out to 8 per row
    act(() => {
      component.root.findByProps({ 'aria-label': 'Zoom Out' }).props.onClick()
    })

    // Attempt to zoom out at min size
    act(() => {
      component.root.findByProps({ 'aria-label': 'Zoom Out' }).props.onClick()
    })

    expect(
      component.root.findByProps({ 'aria-label': 'Zoom Out' }).props.isDisabled,
    ).toBe(true)

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should render empty properly', () => {
    require('next/router').__setUseRouter({
      pathname: '/[projectId]/visualizer',
      query: { projectId: PROJECT_ID },
    })

    require('swr').__setMockUseSWRInfiniteResponse({
      data: [{ count: 0, results: [] }],
    })

    const component = TestRenderer.create(<Assets />)

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should render empty with filter properly', () => {
    const mockRouterPush = jest.fn()

    require('next/router').__setMockPushFunction(mockRouterPush)
    require('next/router').__setUseRouter({
      pathname: '/[projectId]/visualizer',
      query: {
        projectId: PROJECT_ID,
        assetId: ASSET_ID,
        query: encode({
          filters: [
            {
              type: 'similarity',
              attribute: 'analysis.zvi-image-similarity',
              values: { ids: [ASSET_ID] },
            },
          ],
        }),
      },
    })

    require('swr').__setMockUseSWRInfiniteResponse({
      data: [{ count: 0, results: [] }],
    })

    const component = TestRenderer.create(<Assets />)

    expect(component.toJSON()).toMatchSnapshot()

    act(() => {
      component.root
        .findByProps({ children: 'Clear All Filters' })
        .props.onClick()
    })

    expect(mockRouterPush).toHaveBeenCalledWith(
      `/[projectId]/visualizer?assetId=${ASSET_ID}`,
      `/${PROJECT_ID}/visualizer?assetId=${ASSET_ID}`,
    )
  })

  it('should render empty with no filters properly', () => {
    require('next/router').__setUseRouter({
      pathname: '/[projectId]/visualizer',
      query: {
        projectId: PROJECT_ID,
        assetId: ASSET_ID,
      },
    })

    require('swr').__setMockUseSWRInfiniteResponse({
      data: [{ count: 0, results: [] }],
    })

    const component = TestRenderer.create(<Assets />)

    expect(component.toJSON()).toMatchSnapshot()
  })
})
