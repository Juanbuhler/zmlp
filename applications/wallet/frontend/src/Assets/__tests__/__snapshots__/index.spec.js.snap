// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Assets /> should render properly 1`] = `
.emotion-5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0 {
  padding: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Roboto Condensed;
  font-size: 15px;
  line-height: 19px;
  background-color: #252525;
  color: #808080;
}

.emotion-4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  margin-bottom: -2px;
  box-shadow: inset 0 0 4px 0 rgba(0,0,0,0.6);
}

.emotion-3 {
  position: absolute;
  bottom: 16px;
  right: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 2px;
  background-color: #252525;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.5);
  z-index: 10;
  opacity: 0.8;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: not-allowed;
  color: #ffffff;
  background-color: transparent;
  padding: 4px;
  opacity: 1;
  padding: 8px;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-1:hover {
  color: #73b61c;
}

.emotion-1[aria-disabled=true] {
  color: #808080;
}

.emotion-2 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 4px;
  opacity: 1;
  padding: 8px;
}

.emotion-2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-2:hover {
  color: #73b61c;
}

.emotion-2[aria-disabled=true] {
  color: #808080;
}

<div
  className="emotion-5"
>
  <div
    className="emotion-0"
  >
    574
     Assets
  </div>
  <div
    className="emotion-4"
  >
    <div
      style={
        Object {
          "height": 0,
          "overflow": "visible",
          "width": 0,
        }
      }
    />
    <div
      className="emotion-3"
    >
      <button
        aria-disabled={true}
        aria-label="Zoom Out"
        className="emotion-1"
        onClick={[Function]}
        type="button"
      >
        <svg
          data-file-name="circleMinus.svg"
          width={20}
        />
      </button>
      <button
        aria-label="Zoom In"
        className="emotion-2"
        onClick={[Function]}
        type="button"
      >
        <svg
          data-file-name="circlePlus.svg"
          width={20}
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`<Assets /> should render properly 2`] = `
.emotion-5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0 {
  padding: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Roboto Condensed;
  font-size: 15px;
  line-height: 19px;
  background-color: #252525;
  color: #808080;
}

.emotion-4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  margin-bottom: -2px;
  box-shadow: inset 0 0 4px 0 rgba(0,0,0,0.6);
}

.emotion-3 {
  position: absolute;
  bottom: 16px;
  right: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 2px;
  background-color: #252525;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.5);
  z-index: 10;
  opacity: 0.8;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
}

.emotion-2 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: not-allowed;
  color: #ffffff;
  background-color: transparent;
  padding: 4px;
  opacity: 1;
  padding: 8px;
}

.emotion-2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-2:hover {
  color: #73b61c;
}

.emotion-2[aria-disabled=true] {
  color: #808080;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 4px;
  opacity: 1;
  padding: 8px;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-1:hover {
  color: #73b61c;
}

.emotion-1[aria-disabled=true] {
  color: #808080;
}

<div
  className="emotion-5"
>
  <div
    className="emotion-0"
  >
    574
     Assets
  </div>
  <div
    className="emotion-4"
  >
    <div
      style={
        Object {
          "height": 0,
          "overflow": "visible",
          "width": 0,
        }
      }
    />
    <div
      className="emotion-3"
    >
      <button
        aria-label="Zoom Out"
        className="emotion-1"
        onClick={[Function]}
        type="button"
      >
        <svg
          data-file-name="circleMinus.svg"
          width={20}
        />
      </button>
      <button
        aria-disabled={true}
        aria-label="Zoom In"
        className="emotion-2"
        onClick={[Function]}
        type="button"
      >
        <svg
          data-file-name="circlePlus.svg"
          width={20}
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`<Assets /> should render properly 3`] = `
.emotion-5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0 {
  padding: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Roboto Condensed;
  font-size: 15px;
  line-height: 19px;
  background-color: #252525;
  color: #808080;
}

.emotion-4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  margin-bottom: -2px;
  box-shadow: inset 0 0 4px 0 rgba(0,0,0,0.6);
}

.emotion-3 {
  position: absolute;
  bottom: 16px;
  right: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 2px;
  background-color: #252525;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.5);
  z-index: 10;
  opacity: 0.8;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: not-allowed;
  color: #ffffff;
  background-color: transparent;
  padding: 4px;
  opacity: 1;
  padding: 8px;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-1:hover {
  color: #73b61c;
}

.emotion-1[aria-disabled=true] {
  color: #808080;
}

.emotion-2 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 4px;
  opacity: 1;
  padding: 8px;
}

.emotion-2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-2:hover {
  color: #73b61c;
}

.emotion-2[aria-disabled=true] {
  color: #808080;
}

<div
  className="emotion-5"
>
  <div
    className="emotion-0"
  >
    574
     Assets
  </div>
  <div
    className="emotion-4"
  >
    <div
      style={
        Object {
          "height": 0,
          "overflow": "visible",
          "width": 0,
        }
      }
    />
    <div
      className="emotion-3"
    >
      <button
        aria-disabled={true}
        aria-label="Zoom Out"
        className="emotion-1"
        onClick={[Function]}
        type="button"
      >
        <svg
          data-file-name="circleMinus.svg"
          width={20}
        />
      </button>
      <button
        aria-label="Zoom In"
        className="emotion-2"
        onClick={[Function]}
        type="button"
      >
        <svg
          data-file-name="circlePlus.svg"
          width={20}
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`<Assets /> should render properly while loading 1`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  to {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0 {
  padding: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Roboto Condensed;
  font-size: 15px;
  line-height: 19px;
  background-color: #252525;
  color: #808080;
}

.emotion-7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  margin-bottom: -2px;
  box-shadow: inset 0 0 4px 0 rgba(0,0,0,0.6);
}

.emotion-3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
}

.emotion-2 {
  min-height: 300px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  color: #808080;
  background-color: #252525;
  font-size: 15px;
  line-height: 19px;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.5);
}

.emotion-1 {
  -webkit-animation: animation-0 2s linear infinite;
  animation: animation-0 2s linear infinite;
  margin-right: 16px;
}

.emotion-6 {
  position: absolute;
  bottom: 16px;
  right: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 2px;
  background-color: #252525;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.5);
  z-index: 10;
  opacity: 0.8;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
}

.emotion-4 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: not-allowed;
  color: #ffffff;
  background-color: transparent;
  padding: 4px;
  opacity: 1;
  padding: 8px;
}

.emotion-4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-4[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-4:hover {
  color: #73b61c;
}

.emotion-4[aria-disabled=true] {
  color: #808080;
}

.emotion-5 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 4px;
  opacity: 1;
  padding: 8px;
}

.emotion-5:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-5[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-5:hover {
  color: #73b61c;
}

.emotion-5[aria-disabled=true] {
  color: #808080;
}

<div
  className="emotion-8"
>
  <div
    className="emotion-0"
  >
     Assets
  </div>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-3"
    >
      <div
        className="Loading emotion-2"
      >
        <svg
          className="emotion-1"
          data-file-name="generating.svg"
          height={20}
        />
        Loading...
      </div>
    </div>
    <div
      style={
        Object {
          "height": 0,
          "overflow": "visible",
          "width": 0,
        }
      }
    />
    <div
      className="emotion-6"
    >
      <button
        aria-disabled={true}
        aria-label="Zoom Out"
        className="emotion-4"
        onClick={[Function]}
        type="button"
      >
        <svg
          data-file-name="circleMinus.svg"
          width={20}
        />
      </button>
      <button
        aria-label="Zoom In"
        className="emotion-5"
        onClick={[Function]}
        type="button"
      >
        <svg
          data-file-name="circlePlus.svg"
          width={20}
        />
      </button>
    </div>
  </div>
</div>
`;
